spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigServiceTemplate.java

```diff
@@ -50,18 +48,40 @@ public ConfigServiceTemplate(ConfigHttpClient httpClient, ConnectionStringPool c
     }
 
     @Override
-    public List<KeyValueItem> getKeys(String prefix, String storeName, List<String> labels) {
+    public List<KeyValueItem> getKeys(@NonNull String storeName, @NonNull QueryOptions options) {
         Assert.hasText(storeName, "Config store name should not be null or empty.");
+        Assert.notNull(options, "The query options should not be null.");
 
         ConnectionString connString = connectionStringPool.get(storeName);
         String storeEndpoint = connString.getEndpoint();
 
-        String requestUri = new RestAPIBuilder().withEndpoint(storeEndpoint).buildKVApi(prefix, labels);
+        String requestUri = new RestAPIBuilder().withEndpoint(storeEndpoint).buildKVApi(options);
+
+        return getKeys(requestUri, options, storeName);
+    }
+
+    @Override
+    public List<KeyValueItem> getRevisions(@NonNull String storeName, @NonNull QueryOptions options) {
+        Assert.hasText(storeName, "Config store name should not be null or empty.");
+        Assert.notNull(options, "Query options should not be null or empty.");
+
+        ConnectionString connString = connectionStringPool.get(storeName);
+        String storeEndpoint = connString.getEndpoint();
+
+        String requestUri = new RestAPIBuilder().withEndpoint(storeEndpoint).buildRevisionsApi(options);
+
+        return getKeys(requestUri, options, storeName);
+
+    }
+
+    private List<KeyValueItem> getKeys(String requestUri, QueryOptions options, String storeName) {
+        ConnectionString connString = connectionStringPool.get(storeName);
+        String storeEndpoint = connString.getEndpoint();
         List<KeyValueItem> result = new ArrayList<>();
 
         CloseableHttpResponse response = null;
         try {
-            while ((response = getRawResponse(requestUri, connString)) != null) {
+            while ((response = getRawResponse(requestUri, connString, options)) != null) {
```

Comment:
May figure out better layout of this loop, too many if branch in loop I think. We can take care of that later.

Reply:
Will update later.